src/w2grid.js

```diff
@@ -674,6 +674,7 @@
             // process sort
             this.records.sort(function (a, b) {
                 var ret = 0;
+                var defaultToString = {}.toString;
```

Comment:
Ok, sorry. If you look at the file I just commited below (test/grid4.xhtml), and run it with an
unmodified version of w2ui, you will see weird behaviour if you type 'undefined' or 'object' or '[' in
the search box:

![screen shot 2015-08-10 at 14 04 30](https://cloud.githubusercontent.com/assets/7224918/9171050/c0c00ba8-3f68-11e5-9ce4-fdd3f1f355ca.png)
![screen shot 2015-08-10 at 14 05 34](https://cloud.githubusercontent.com/assets/7224918/9171073/e47f1f70-3f68-11e5-83b2-fc987b4772b4.png)

This is because of the way variables are converted to String for searching:
 - Missing fields are converted to "undefined",
 - The default toString() method convert objects into the string: "[object Object]"

So basically, when searching:
  1)  we don't want to match Objects that do not redefine the standard toString()
        method, so we need to skip it if obj.toString = {}.toString.
  2) But we DO want to match Objects that have a better toString method, like
       our custom objects or, for example, the Date object.

Anyway, I'm still not happy with the sorting function, I will add a few other patches.